### PR TITLE
[tests-only] Add unit tests for the text editor

### PR DESCRIPTION
**Pull request by JammingBen**
This also includes a mock for the `useAppDefaults` composable.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

### CODE REVIEW

**Comment by JammingBen** [comment]
SonarCloud fails because of 0.0% condition coverage... makes sense 🤡 
image

**Review by dschmidt — APPROVED** [review]
I like, can we add this to stable instead?

File: packages/web-app-text-editor/tests/unit/app.spec.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+function getWrapper({ fileName = 'someFile.txt' }: { fileName?: string } = {}) {
+  jest.mocked(useAppDefaults).mockImplementation(() =>
+    useAppDefaultsMock({
+      currentFileContext: ref(mockDeep<FileContext>({ path: fileName }))
+    })
+  )
+  const defaultMocks = { ...defaultComponentMocks() }
+  const storeOptions = { ...defaultStoreMockOptions }
+  const localVue = defaultLocalVue()
+  const store = createStore(Vuex.Store, storeOptions)
```
Comment (dschmidt) [inline packages/web-app-text-editor/tests/unit/app.spec.ts:49]: Out of scope for this PR, but:

I don't like that you use options from test helpers but have to add `vuex-extensions` to the package dependencies, I would like to see this handled in `web-test-helpers` completely.
We should really throw some time at unit tests at some point, unify the setup and clean up stuff like this.

File: packages/web-test-helpers/src/mocks/useAppDefaultsMock.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { mockDeep } from 'jest-mock-extended'
+import { ref } from '@vue/composition-api'
+import { Resource } from 'web-client/src'
+import { FileResource } from 'web-client/src/helpers'
+import { GetFileContentsResponse } from 'web-client/src/webdav/getFileContents'
+import { AppConfigObject, FileContext, useAppDefaults } from 'web-pkg/src'
+
+export const useAppDefaultsMock = (
```
Comment (kulmann) [inline packages/web-test-helpers/src/mocks/useAppDefaultsMock.ts:8]: Still amazed by the mocks with the partial type for options 😎

**Review by dschmidt — APPROVED** [review]

File: packages/web-app-text-editor/tests/unit/app.spec.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      expect(wrapper.find('oc-spinner-stub').exists()).toBeTruthy()
+    })
+    it('shows the editor after loading', async () => {
+      const { wrapper } = getWrapper()
+      await wrapper.vm.loadFileTask.last
+      expect(wrapper.find('oc-spinner-stub').exists()).toBeFalsy()
+      expect(wrapper.find('oc-textarea-stub').exists()).toBeTruthy()
+    })
+  })
+  it('shows the preview', async () => {
```
Comment (kulmann) [inline packages/web-app-text-editor/tests/unit/app.spec.ts:33]: Could you make this a `describe` block about previews and then check that different file types show/don't show the preview? The preview is only there for markdown. E.g. a `.txt` file will not have a preview.

**Review by kulmann — CHANGES_REQUESTED** [review]

**Review by kulmann — APPROVED** [review]

**Pull request merged**